File: frontend/cart-frontend/src/service/CartService.js
import axios from "axios";

const API_BASE_URL = process.env.BFF_URL;

export const getCartsByUserId = async (userId, page = 0, size = 3) => {
  try {
    const response = await axios.get(`${API_BASE_URL}/carts/user/${userId}`, {
      params: { page, size },
    });
    //console.log("Cart Items:", response.data);
    return response.data;
  } catch (error) {
    console.error("Error fetching carts:", error);
    throw new Error("Error fetching carts");
  }
};

export const getCartById = async (cartId) => {
  const response = await axios.get(`${API_BASE_URL}/carts/${cartId}`);
  return response.data;
};

export const createCart = async (cart) => {
  const response = await axios.post(`${API_BASE_URL}/carts`, cart);
  return response.data;
};

export const deleteCart = async (cartId) => {
  await axios.delete(`${API_BASE_URL}/carts/${cartId}`);
};

export const updateCart = async (cartId, cart) => {
  const response = await axios.put(`${API_BASE_URL}/carts/${cartId}`, cart);
  return response.data;
};

File: backend/service-carts/src/main/java/com/onlinestore/servicecarts/service/CartService.java
package com.onlinestore.servicecarts.service;

import com.onlinestore.servicecarts.dto.CartDTO;
import com.onlinestore.servicecarts.model.Cart;
import com.onlinestore.servicecarts.repo.CartRepo;
import org.modelmapper.ModelMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.stereotype.Service;

@Service
public class CartService {
    @Autowired
    private CartRepo cartRepository;

    @Autowired
    private ModelMapper modelMapper;

    public Page<CartDTO> getCartsByUserId(String userId, PageRequest pageRequest) {
        Page<Cart> carts = cartRepository.findByUserId(userId, pageRequest);
        return carts.map(cart -> modelMapper.map(cart, CartDTO.class));
    }

    public CartDTO getCartById(String cartId) {
        Cart cart = cartRepository.findById(cartId).orElse(null);
        return modelMapper.map(cart, CartDTO.class);
    }

    public CartDTO createCart(CartDTO cartDto) {
        Cart cart = modelMapper.map(cartDto, Cart.class);
        cart = cartRepository.save(cart);
        return modelMapper.map(cart, CartDTO.class);
    }

    public void deleteCart(String cartId) {
        cartRepository.deleteById(cartId);
    }

    public CartDTO updateCart(String cartId, CartDTO cartDto) {
        Cart cart = modelMapper.map(cartDto, Cart.class);
        cart.setCartId(cartId);
        cart = cartRepository.save(cart);
        return modelMapper.map(cart, CartDTO.class);
    }
}